### PR TITLE
Adjustable Refresh Rate

### PR DESCRIPTION
**Pull request by Yayroos**
Add a setting in Advanced Settings that allows the polling rate of currently playing and liked to be set.

### CODE REVIEW

**Comment by Yayroos** [comment]
#308 #177 #234 #201 - fix and/or workaround for all these (any any others about the API Throttling/constant calls. Allows the frequency to be set in Advanced Settings to your preference (default is current behaviour, currently playing every second)

File: src/renderer/windows/settings/advanced-settings.tsx
Comment (stamoun) [inline src/renderer/windows/settings/advanced-settings.tsx:30]: Don't use milliseconds here, use seconds instead

```suggestion
            Refresh Time (seconds)
```

File: src/renderer/windows/settings/advanced-settings.tsx
Comment (stamoun) [inline src/renderer/windows/settings/advanced-settings.tsx:33]:
```suggestion
              min={1}
```

File: src/renderer/windows/settings/advanced-settings.tsx
Comment (stamoun) [inline src/renderer/windows/settings/advanced-settings.tsx:34]:
```suggestion
              max={10}
```

File: src/renderer/windows/settings/advanced-settings.tsx
Comment (stamoun) [inline src/renderer/windows/settings/advanced-settings.tsx:35]: Do we really need half a second precision on track refreshes?

```suggestion
              step={1}
```

**Comment by stamoun** [comment]
Please do not include styling change in an MR, follow the current ones (e.g. use single quotes and not double quotes).

Styling change could be done in a separate MR through an automatic process (much easier to review)

**Comment by Yayroos** [comment]
Will do - friend was helping me sort out the git stuff and applied the prettier formatting defaults so everything got rearranged.

**Comment by stamoun** [comment]
> Will do - friend was helping me sort out the git stuff and applied the prettier formatting defaults so everything got rearranged.

No worries, I need a more exhaustive eslint config file (including prettier styling)

**Comment by Yayroos** [comment]
Have reverted all those other files - this should just be the files i touched to actually add the feature

File: src/renderer/windows/settings/advanced-settings.tsx
Comment (Yayroos) [reply src/renderer/windows/settings/advanced-settings.tsx:35]: step={1} would be millisecond precision. We can go to step 1000 if you'd rather, i just picked it pretty much at random.

Comment (Yayroos) [reply src/renderer/windows/settings/advanced-settings.tsx:35]: actually you make a good point. I'll go make that seconds now. Yesterday brain was not very switched on.

File: src/models/settings.ts
Comment (stamoun) [inline src/models/settings.ts:70]: Actually, you can just store in seconds... so this would become (note the variable name)

```suggestion
  trackInfoRefreshTimeInSeconds: 1,
```

File: src/renderer/app/cover/index.tsx
Comment (stamoun) [inline src/renderer/app/cover/index.tsx:182]:
```suggestion
    const listeningToIntervalId = setInterval(handlePlaybackChanged, refreshTime * ONE_SECOND_IN_MS);
```

File: src/renderer/app/cover/index.tsx
Comment (stamoun) [inline src/renderer/app/cover/index.tsx:192]:
```suggestion
    const refreshTrackLikedIntervalId = setInterval(refreshTrackLiked, 2 * refreshTime * ONE_SECOND_IN_MS);
```

**Review by stamoun — CHANGES_REQUESTED** [review]

**Comment by stamoun** [comment]
Good stuff, I'll let you merge it @Yayroos 🚀

**Comment by stamoun** [comment]
Forgot to add something but I'm sure it's fine...

When you modify the UI in any way, please post a screenshot of your modification (the extra setting in this case). You can still go ahead and merge this PR though.

**Comment by Yayroos** [comment]
i'll grab a screenshot today!

thanks for your quick responses and patience.

**Comment by stamoun** [comment]
> i'll grab a screenshot today!
> 
> thanks for your quick responses and patience.

No worries, it's fun to see contributions to Lofi :) Also, thanks for being understanding. I hope you learned a bit during this review.

**Comment by Yayroos** [comment]
i will get to this - i got covid and sitting up for more than 5 minutes hurts

**Comment by stamoun** [comment]
> i will get to this - i got covid and sitting up for more than 5 minutes hurts

No worries, I can squash and merge for you, but I want you to press that button. I don't know why but it's a good feeling to press that `Squash and merge` button :)

This code isn't going anywhere.

**Comment by Yayroos** [comment]
i'll do it, maybe this afternoon if i can actually use my proper laptop. I'm keen to do it, i just can't think straight or sit up properly at the moment, so it'll have to wait until i shake the damn virus

**Comment by Yayroos** [comment]
lofi new setting refresh time
screenshot of the new setting :)

**Comment by Yayroos** [comment]
i dont seem to actually have access to merge it, at the very least i cant see any button indicating to do so

**Comment by stamoun** [comment]
Could you change the option label to something like `API polling interval (Seconds)`?

Don’t worry about merging it, I thought PR authors could merge once approved, seems like not.

I’ll merge it early next week.

**Review by stamoun — APPROVED** [review]

**Comment by stamoun** [comment]
That was fast! Merging! 🚀

**Comment by Yayroos** [comment]
have changed the setting label. thanks for the patience. still absolutely rolled by covid at the moment.

**Comment by ccxxv** [comment]
> lofi new setting refresh time screenshot of the new setting :)

How do I add this to mine? I don't seem to have this option.

**Comment by Yayroos** [comment]
> 
> How do I add this to mine? I don't seem to have this option.

doesn't look like anyone's generated a release with this in - I'll try in the next couple of days on my fork if i can figure it out, else you can build it from source if you can install all the stuff.

**Pull request merged**
